### ISSUE TITLE
Can't fill the screen with the browser

### ISSUE THREAD

**Issue by AdamBark** [chadparry/kodi-plugin.program.remote.control.browser#3]
My browser opens up but does not fill the screen. If I press F11 to full-screen the browser it seems to work but it is still cropped to the original window size leaving half the screen still filled with kodi.

**Comment by AdamBark** [comment]
I managed to manually set the correct window size using 'xdotool setwindowsize' by trying all of the window ids that show up for firefox.

**Comment by chadparry** [comment]
Thanks for the report. Which OS are you using? Do you know whether the problem affects all browsers for you?

**Comment by AdamBark** [comment]
Thanks for the addon. Ubuntu 16.04, I tried firefox and chromium and neither uses the full screen by default. I am not running a  window manager of any sort just raw X with kodi, would you expect that to work?

**Comment by chadparry** [comment]
[Sorry for the slow response.]

I also tried to run Kodi without a window manager for a while. Eventually I gave up, because of problems like this. There were comments on the forums about how running Kodi without a window manager was unreliable and not supported, (cf. https://askubuntu.com/a/1012501). Based on my experience, I have to agree. Besides the problem of the browser not using the full screen, sometimes it would get hidden behind Kodi. I wrote a workaround (https://github.com/chadparry/kodi-plugin.program.remote.control.browser/commit/0b7306664c4e1fed851f573f6a18ca7c78789efe) that you can check out. There's Python code there to detect the window ID for the correct browser instance, and then to use xdotool to raise it. Even with that workaround present, the behavior was inconsistent. It bugged me because it only worked _most_ of the time. After I had tried everything I could think of, I removed that workaround and just started running Kodi with a window manager. I also realized that the browser itself requires a window manager anyway. Window managers like LXDE require so few resources that there's little reason not to run one.

So, in summary, I think your suspicion about no window manager is correct, and I recommend that you try running LXDE. I believe that your full screen problem, and other bugs besides that I have seen, will all be resolved.

If you have a use case that doesn't allow you to run a window manager, then I can revisit that workaround I linked, and see whether it can be made to work more reliably.

**Comment by AdamBark** [comment]
Thanks for getting back to me. I installed openbox and it fixed my similar problem with Rom Collection Browser as well so I think it makes sense to do that. I'm not sure how libreelec deals with it though, perhaps they just don't bother.
